engine_modules/corporation/models.py

```diff
@@ -113,7 +141,14 @@ def update(corporation, delta):
 				except Corporation.DoesNotExist:
 					return
 
-			corporation.update_assets(delta, category=delta_category)
+			if market is None:
+				# By default, a random market is impacted
+				market = corporation.random_market
```

Comment:
thinking back on this---if we want a random market, at least we should make it a function.

A property hides the fact that there is code behind, sometime it's cool but for randomness it's unclear, especially since

a.random_market != a.random_market